src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -64,6 +64,26 @@ inline auto strlen = [](std::optional<std::string> s) {
 };
 using StrlenExpression = StringExpressionImpl<1, decltype(strlen)>;
 
+// LCASE
+static auto lowercaseImpl = [](std::optional<std::string> input) -> IdOrString {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  } else {
```

Comment:
A cascaded question: When exactly is the argument `std.:nullopt`, when the argument given in the SPARQL query was not actually a string? If yes, is the result of a built-in SPARQL function taking one or several string as argument always UNDEF when one of these arguments is not a string? If yes, should this be handled in the functions or outside? I am asking because for the numeric functions the `return Id::makeUndefined` happens elsewhere.

Reply:
This is actually a good idea.
For the Numeric functions we have a wrapper that handles the "invalid input" -> expression error -> UNDEF result case automagically, 
But we could extend this to all kinds of expressions, s.t. for example the string function implementations only need to handle a `string` instead of an `std::optional<std::string>`.
I will think of something and integrate it into one of the next PRs.